compass/ocean/tests/nonhydro/stratified_seiche/initial_state.py

```diff
@@ -0,0 +1,221 @@
+        vertical_coordinate = 'uniform'
+
+        ds = dsMesh.copy()
+        nCells = ds.nCells.size
+        nEdges = ds.nEdges.size
+        nVertices = ds.nVertices.size
+
+        xCell = ds.xCell
+        yCell = ds.yCell
+        xEdge = ds.xEdge
+        yEdge = ds.yEdge
+        angleEdge = ds.angleEdge
+        cellsOnEdge = ds.cellsOnEdge
+        edgesOnCell = ds.edgesOnCell 
+
+        # Adjust coordinates so first edge is at zero in x and y
+        xOffset = xEdge.min()
+        xCell -= xOffset
+        xEdge -= xOffset
+        yOffset = np.min(yEdge)
+        yCell -= yOffset
+        yEdge -= yOffset
+
+        # x values for convenience
+        xCellMax = max(xCell)
+        xCellMin = min(xCell)
+        xCellMid = 0.5 * (xCellMax + xCellMin)
+        xEdgeMax = max(xEdge)
+
+        # initialize velocity field 
+        u = np.zeros([1, nEdges, nVertLevels])
+
+        # comment('create and initialize variables')
+        time1 = time.time()
+
+        varsZ = ['refLayerThickness', 'refBottomDepth', 'refZMid',
+            'vertCoordMovementWeights']
+        for var in varsZ:
+            globals()[var] = np.nan * np.ones(nVertLevels)
+
+        vars2D = ['ssh', 'bottomDepth', 'surfaceStress',
+            'atmosphericPressure', 'boundaryLayerDepth']
+        for var in vars2D:
+            globals()[var] = np.nan * np.ones(nCells)
+        maxLevelCell = np.ones(nCells, dtype=np.int32)
+
+        vars3D = [ 'layerThickness', 'temperature', 'salinity',
+            'zMid', 'density']
+        for var in vars3D:
+            globals()[var] = np.nan * np.ones([1, nCells, nVertLevels])
+        restingThickness = np.nan * np.ones([nCells, nVertLevels])
+
+        # Note that this line shouldn't be required, but if layerThickness is
+        # initialized with nans, the simulation dies. It must multiply by a
+        # a nan on a land cell on an edge, and then multiply by zero.
+        layerThickness[:] = -1e34
+
+        # equally spaced layers
+        refLayerThickness[:] = maxDepth / nVertLevels
+        refBottomDepth[0] = refLayerThickness[0]
+        refZMid[0] = -0.5 * refLayerThickness[0]
+        for k in range(1, nVertLevels):
+            refBottomDepth[k] = refBottomDepth[k - 1] + refLayerThickness[k]
+            refZMid[k] = -refBottomDepth[k - 1] - 0.5 * refLayerThickness[k]
+
+        # SSH
+        ssh[:] = 0.0
+
+        # Compute maxLevelCell and layerThickness for z-level
+        # (variation only on top)
+        if (vertical_coordinate == 'z'):
+            vertCoordMovementWeights[:] = 0.0
+            vertCoordMovementWeights[0] = 1.0
+            for iCell in range(0, nCells):
+                maxLevelCell[iCell] = nVertLevels - 1
+                bottomDepth[iCell] = refBottomDepth[nVertLevels - 1] 
+                layerThickness[0, iCell, :] = refLayerThickness[:]
+                layerThickness[0, iCell, 0] += ssh[iCell]
+            restingThickness[:, :] = layerThickness[0, :, :]
```

Comment:
I don't believe we have such a coordinated currently.  What we call z-level is a bit different in that it drops layers to accommodate the SSH (useful for handling ice topography over ice-shelf cavities) rather than having all the variation in the top layer.

It seems better, though, to add this alternative coordinate to the existing vertical coordinate framework rather than just having it here.  We should talk about how to add that, what to call it, etc.

Reply:
The option
`vertical_coordinate == 'z'`
corresponds to the z-level vertical coordinate system where the SSH perturbations are placed all in the
top level. In the MPAS namelist this would be 
`config_vert_coord_movement = 'fixed'`
The other option 
`vertical_coordinate == 'uniform'`
corresponds to the z-star vertical coordinate system, which in the MPAS namelist would be
`config_vert_coord_movement = 'uniform_stretching'`
I wanted to give both options in the python script.

Reply:
Okay, let's leave this part of the code as it is.  I would be glad to follow up with some clean-up that would move this coordinate to the shared infrastructure at a later date.

Reply:
I got rid of this part of code since now there is
`init_vertical_coord(config, ds)` .